tests/core/e2e/test_e2e_finetune_hydra.py

```diff
@@ -0,0 +1,194 @@
+    ft_config = {
+        "mode": FineTuneMode.RETAIN_BACKBONE_ONLY.name,
+        "starting_checkpoint": starting_ckpt,
+        "heads": {
+            "energy": {
+                "module": "equiformer_v2_energy_head"
+            },
+            "forces": {
+                "module": "equiformer_v2_force_head"
+            }
+        }
+    }
+    run_main_with_ft_hydra(tempdir = tempdir2,
+                           yaml = ft_yml,
+                           data_src = tutorial_val_src,
+                           run_args = {"seed": 1000},
+                           ft_config = ft_config,
+                           output_checkpoint = ck_ft_path)
+    assert os.path.isfile(ck_ft_path)
+    ft_ckpt = torch.load(ck_ft_path)
+    assert "config" in ft_ckpt
+    assert ft_ckpt["config"]["model"]["name"] == FTHYDRA_NAME
+    # check that the backbone weights are the same, and other weights are not the same
+    new_state_dict = ft_ckpt["state_dict"]
+    for key in new_state_dict:
+        if key.startswith("backbone"):
+            # backbone should be identical
+            assert torch.allclose(new_state_dict[key], old_state_dict[key])
+        elif key.startswith("output_heads") and key.endswith("weight"):
+            # heads weight should be different because the seeds are different
+            assert not torch.allclose(new_state_dict[key], old_state_dict[key])
+
+
+def test_finetune_hydra_data_only(tutorial_val_src):
+    starting_ckpt, original_tmpdir = make_checkpoint(tutorial_val_src, 0)
+    old_state_dict = torch.load(starting_ckpt)["state_dict"]
+    # now finetune a the model with the checkpoint from the first job
+    tempdir2 = tempfile.TemporaryDirectory()
+    ft_yml = Path("tests/core/models/test_configs/test_finetune_hydra.yml")
+    ck_ft_path = os.path.join(tempdir2.name, "checkpoint_ft.pt")
+    ft_config = {
+        "mode": FineTuneMode.DATA_ONLY.name,
+        "starting_checkpoint": starting_ckpt,
+    }
+    run_main_with_ft_hydra(tempdir = tempdir2,
+                           yaml = ft_yml,
+                           data_src = tutorial_val_src,
+                           run_args = {"seed": 1000},
+                           ft_config = ft_config,
+                           output_checkpoint = ck_ft_path)
+    assert os.path.isfile(ck_ft_path)
+    ft_ckpt = torch.load(ck_ft_path)
+    assert "config" in ft_ckpt
+    config_model = ft_ckpt["config"]["model"]
+    assert config_model["name"] == FTHYDRA_NAME
+    # check that the entire model weights are the same
+    new_state_dict = ft_ckpt["state_dict"]
+    assert len(new_state_dict) == len(old_state_dict)
+    for key in new_state_dict:
+        assert torch.allclose(new_state_dict[key], old_state_dict[key])
+    # check the new checkpoint contains a hydra model
+    assert FTConfig.STARTING_MODEL in config_model[FTConfig.FT_CONFIG_NAME]
+
+
+def test_finetune_from_finetunehydra(tutorial_val_src):
```

Comment:
Would it be worth while to have a test that runs a few iterations of Finetune Hydra and overfits on the training set? Just to make sure that Finetune hydra can correctly learn something?
Its a bit loose test, but just any training error < X or < that of original checkpoint?
Might be for BE week!

Reply:
ya i think since this doesnt do anything special, i thought just checking the behavior that it loaded the weights correctly is sufficient, we need to do some work to get a finetune setting to have it learn something